Sources/PrinceOfVersions/Objective-C Helpers/Objective-C Extensions/CheckUpdateFromAppStoreObjectiveCExtensions.swift

```diff
@@ -0,0 +1,117 @@
+    @discardableResult
+    public static func checkForUpdateFromAppStore(
+        trackPhaseRelease: Bool,
+        completion: @escaping AppStoreObjectCompletionBlock,
+        error: @escaping ObjectErrorBlock
+    ) -> URLSessionDataTask? {
+        return internalyCheckAndPrepareForUpdateAppStore(bundle: .main, trackPhaseRelease: trackPhaseRelease, callbackQueue: .main, completion: completion, error: error)
+    }
+
+    /**
+     Used for getting the versioning information from the AppStore Connect.
+
+     After check with server is finished, this method will return all informations about the app versioning available on AppStore Connect.
+     It's up to the user to handle that info in a way sutable for the app.
+
+     Update status can only take on value `UpdateStatus.noUpdateAvailable` or `UpdateStatus.newUpdateAvailable`, but it can't be `UpdateStatus.requiredUpdateNeeded` since there is no way to determine if the update version is mandatory with this check.
+
+     If flag `trackPhaseRelease` is set to `NO`, the value of the `phaseReleaseInProgress` will instantly be `NO` as phased release is not used.
+     Otherwise, if we have to check `trackPhaseRelease`, value of `phaseReleaseInProgress` will return `NO` once phased release period of 7 days is over.
+
+     __WARNING:__ As we are not able to determine if phased release period is finished earlier (release to all options is selected after a while), if `trackPhaseRelease` is enabled `phaseReleaseInProgress` will return `NO` only after 7 days of `currentVersionReleaseDate` value set on AppStore Connect.
+
+     If parameter `notificationFrequency` is set to `.always`  and latest version of the app is bigger than installed version, method will always return `.newUpdateAvailable`. However, if the`notificationFrequency` is set to `.once`, only first time this method is called for the same latest app version, it will return `.newUpdateAvailable`, each subsequent call, it will return `.noUpdateAvailable`.
+
+     - Parameters:
+        * bundle: Bundle where .plist file is stored in which app identifier and app versions should be checked.
+        * trackPhaseRelease: Boolean that indicates whether PoV should notify about new version after 7 days when app is fully rolled out or immediately. Default value is `YES`.
+        * callbackQueue: The queue on which the completion handler is dispatched. By default, `main` queue is used.
+        * notificationFrequency: Determines update status appearance frequency.
+        * completion: The completion handler to call when the load request is complete. It returns result that contains UpdatInfo data or PoVError error
+
+     - Returns:
+        * Discardable `URLSessionDataTask`
+     */
+    @available(swift, obsoleted: 1.0)
+    @objc(checkForUpdateFromAppStoreWithTrackPhasedRelease:callbackQueue:bundle:notificationFrequency:completion:error:)
+    @discardableResult
+    public static func checkForUpdateFromAppStore(
```

Comment:
Since we are already using the options parameter but with a different purpose in `checkForUpdates` method, there are three possible solutions to this comment:

1. Fill current `PoVRequestOptions` class with all these properties and use one grand options parameter for every method call (including both `checkForUpdates` and `checkForUpdatesFromAppStore`). I'm not fond of this since not all options parameters would be used in every method, and users might think they would. It could cause confusion.

2. We could create another OptionsObjC and fill it with all these parameters. Still, I think this would also cause confusion since options parameter is used in `checkForUpdates`, and method call would have two different options (`PoVRequestOption` and `OptionsObjC`). However, we could create AppStoreOptions and this would solve the issue described in this comment, but it would not solve the issue regarding multiple declarations of `checkForUpdates` method.

3. In this option, I would leave everything as is. There are overloads for most important and most used parameters, for everything else, there is one method declaration which developers can use if they need specific parameters. This is the option I am leaning toward.

Please let me know what you think and if you have any other suggestions other than the three options described. @Truba
cc: @jabou

Reply:
I would leave this as is. Initially, we created one Options object to rule them all (except URL) and that one becomes too crowded and confusing what is request and what is the response property. 

One thing that we can reduce is the number of methods in `CheckUpdatesObjectiveCExtensions` by one and just have this:
* checkForUpdatesFromURL:completion:error: (most common)
* checkForUpdatesFromURL:options:completion:error: (second most common)
* checkForUpdatesFromURL:callbackQueue:options:completion:error: (all options)

Reply:
👍